ivy.el

```diff
@@ -1522,7 +1522,8 @@ See also `ivy-sort-max-size'."
 
 (defun ivy--sort-function (collection)
   "Retrieve sort function for COLLECTION from `ivy-sort-functions-alist'."
-  (cdr (assoc collection ivy-sort-functions-alist)))
+  (or (cdr (assoc collection ivy-sort-functions-alist))
+      (cdr (assoc t ivy-sort-functions-alist))))
```

Comment:
Shouldn't the result of the first `assq` (no need for `assoc`) be checked for `nil` before falling through to the default case? Otherwise disabling sorting for a given collection by setting its sort function to `nil` wouldn't work. What I mean is:

```el
(defun ivy--sort-function (collection)
  "Retrieve sort function for COLLECTION from `ivy-sort-functions-alist'."
  (let ((entry (assq collection ivy-sort-functions-alist)))
    (if entry
        (cdr entry)
      (cdr (assq t ivy-sort-functions-alist)))))
```